nifi-nar-bundles/nifi-jslt-bundle/nifi-jslt-processors/src/main/java/org/apache/nifi/processors/jslt/JSLTTransformJSON.java

```diff
@@ -122,11 +139,18 @@ public class JSLTTransformJSON extends AbstractProcessor {
     private static final List<PropertyDescriptor> descriptors;
     private static final Set<Relationship> relationships;
     private static final ObjectMapper jsonObjectMapper = new ObjectMapper();
+    private static final ObjectMapper jsonObjectMapperPerObject = new ObjectMapper();
+
+    private static final JsonFactory jsonFactory = new JsonFactory();
+    private JsonParser jsonParser;
+    private JsonNode firstJsonNode;
+    private boolean firstObjectConsumed = false;
```

Comment:
This seems like it could introduce thread-safety issues with multiple concurrent tasks. These first JSON Node and object consumed properties should be local to methods and passed around.